### PR TITLE
Fix host deserialization in inventory groups.

### PR DESCRIPTION
**Pull request by mattclay**
##### SUMMARY

Fix host deserialization in inventory groups.

##### ISSUE TYPE

Bugfix Pull Request

##### COMPONENT NAME

inventory

##### ANSIBLE VERSION

```
ansible 2.5.0 (deserialization-fix fa679a10a1) last updated 2017/09/28 01:31:44 (GMT -700)
  config file = None
  configured module search path = [u'/Users/mclay/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/mclay/code/mattclay/ansible/lib/ansible
  executable location = /Users/mclay/code/mattclay/ansible/bin/ansible
  python version = 2.7.13 (default, Jul 18 2017, 09:17:00) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
```

### CODE REVIEW

**Comment by bcoca** [comment]
shipit

**Comment by mattclay** [comment]
This should resolve the hangs we've been seeing in CI as well as https://github.com/ansible/ansible/issues/30903.

File: lib/ansible/inventory/group.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def deserialize(self, data):
         self.name = data.get('name')
         self.vars = data.get('vars', dict())
         self.depth = data.get('depth', 0)
-        self.hosts = data.get('hosts', {})
+        self.hosts = data.get('hosts', [])
```
Comment (abadger) [inline lib/ansible/inventory/group.py:75]: Could do self.hosts = data.get('hosts', tuple())  if this should actually be read-only.

Comment (mattclay) [reply lib/ansible/inventory/group.py:75]: It isn't supposed to be read-only.

**Comment by willthames** [comment]
@mattclay, thanks, this fixes my proper playbook too :)

File: lib/ansible/inventory/group.py (hunk 1 above)
Comment (bcoca) [reply lib/ansible/inventory/group.py:75]: no, its a list, this is where we ADD/REMOVE hosts to a group

**Comment by mattclay** [comment]
Closing in favor of bcoca's fix in https://github.com/ansible/ansible/pull/31111.

**Pull request closed without merging**